lms/djangoapps/courseware/tests/test_grades.py

```diff
@@ -121,3 +127,80 @@ def _gradesets_and_errors_for(self, course_id, students):
                 students_to_errors[student] = err_msg
 
         return students_to_gradesets, students_to_errors
+
+
+class TestMaxScoresCache(ModuleStoreTestCase):
+    """
+    Tests for the MaxScoresCache
+    """
+    def setUp(self):
+        super(TestMaxScoresCache, self).setUp()
+        self.student = UserFactory.create()
+        self.course = CourseFactory.create()
+        self.problems = []
+        for _ in xrange(3):
+            self.problems.append(
+                ItemFactory.create(category='problem', parent=self.course)
+            )
+
+        CourseEnrollment.enroll(self.student, self.course.id)
+        self.request = RequestFactory().get('/')
+        self.locations = [problem.location for problem in self.problems]
+
+    @override_settings(KEY_PREFIX=str(uuid.uuid4()))  # hack to ensure unique cache per test
```

Comment:
instead of doing this, we can just  explicitly  remove any items we added to the cache after the test runs. Not a great solution, but better than overriding the `KEY_PREFIX`.